internal/hcserror/hcserror.go

```diff
@@ -32,18 +32,18 @@ func (e *HcsError) Error() string {
 
 func New(err error, title, rest string) error {
 	// Pass through DLL errors directly since they do not originate from HCS.
-	if _, ok := err.(*syscall.DLLError); ok {
+	if t := (&windows.DLLError{}); errors.As(err, &t) {
```

Comment:
Nit: Assignment in the if for these errors.As cases feels very odd to me to read. I kinda like how they have it laid out in the package docs.
```go
var t *windows.DLLError
if errors.As(err, &t) {
     // Do stuff
}
```

Reply:
And this is from someone who loves doing the below 😆
```go 
if err := thing(); err != nil {

}
```

Reply:
yeah, in hindsight, it looks better as you suggested....